lte/gateway/c/oai/tasks/amf/amf_as.cpp

```diff
@@ -0,0 +1,1003 @@
+      }
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->security_protected.plain.amf);
+    } else {
+      /*
+       * No valid 5G CN security context exists
+       */
+      msg->header.security_header_type = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->plain.amf);
+    }
+  } else {
+    /*
+     * No valid 5G CN security context exists
+     */
+    msg->header.security_header_type = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->plain.amf);
+  }
+
+  /*
+   * A valid 5G CN security context exists but NAS integrity key
+   * * * * is not available
+   */
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, NULL);
+}
+
+/***************************************************************************
+ **                                                                       **
+ ** Name:        amf_send_identity_request()                              **
+ **                                                                       **
+ ** Description: Send Identity Request to UE                              **
+ **                                                                       **
+ ** Inputs:      msg: Security msg                                        **
+ **              amf_msg :   amf msg                                      **
+ **                                                                       **
+ ** Return:   size                                                        **
+ **                                                                       **
+ **************************************************************************/
+static int amf_send_identity_request(
+    const amf_as_security_t* msg, IdentityRequestMsg* amf_msg) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int size = AMF_HEADER_MAXIMUM_LENGTH;
+  /*
+   * Mandatory - Message type
+   */
+  amf_msg->message_type.msg_type  = IDENTITY_REQUEST;
+  amf_msg->m5gs_identity_type.toi = M5G_IDENTITY_SUCI;
+  size += IDENTITY_TYPE_2_IE_MAX_LENGTH;
+  if (msg->ident_type == IDENTITY_TYPE_2_IMSI) {
+    amf_msg->m5gs_identity_type.toi = IDENTITY_TYPE_2_IMSI;
+  }
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, size);
+}
+
+static void _s6a_handle_authentication_info_ans(
+    const std::string& imsi, uint8_t imsi_length, const grpc::Status& status,
+    feg::AuthenticationInformationAnswer response, s6a_auth_info_ans_t* aia_p) {
+  strncpy(aia_p->imsi, imsi.c_str(), imsi_length);
+  aia_p->imsi_length = imsi_length;
+
+  if (status.ok()) {
+    if (response.error_code() < feg::ErrorCode::COMMAND_UNSUPORTED) {
+      magma::convert_proto_msg_to_itti_s6a_auth_info_ans(response, aia_p);
+      OAILOG_DEBUG(
+          LOG_AMF_APP,
+          "Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: %s", imsi);
+    }
+  } else {
+    OAILOG_INFO(
+        LOG_AMF_APP,
+        "S6A-AUTHENTICATION_INFORMATION_ANSWER failed with "
+        "status:%d, StatusCode:%d\n",
+        status.error_message(), response.error_code());
+  }
+}
```

Comment:
Why is the S6a function here? Can we not use it from the existing S6a client api?

Reply:
Will be taken care in upcoming PR